### ISSUE TITLE
Lightweight server dispatchers

### ISSUE THREAD

**Issue by g0dm0d3** [HangfireIO/Hangfire#2461]
Hi.
We're using multiple servers (1 **full** + N **lightweight**),
and it looks like that there are not enough settings for flexible configuration of a lightweight server.

For example: we want to turn on **SqlServerHeartbeatProcess** dispatcher (so that jobs can run longer than **SlidingInvisibilityTimeout**),
but at the same time we want **RecurringJobScheduler/DelayedJobScheduler** to be turned on a full server only.

Currently there is single setting **BackgroundJobServerOptions.IsLightweightServer** which turns on storage-related processes AND recurring/delayed job schedulers,

can it be split into multiple settings ?

Thanks!

**Comment by davewebscale** [comment]
We have a similar problem/requirement.
For us the problem is that we have SQL storage in one region, but have some servers on the other side of the world hitting the same storage (perhaps this is inadvisable).

We have found that the servers running far away have high latency when enqueuing jobs only managing a handful per second. When a cross region server is taking responsibility for storage processing and there are a lot of jobs are scheduled in a burst the system would fall behind.

I haven't worked out why, but these cross region servers seem always be the ones doing the storage work. I assume because they are so much slower there is always more to do when they get through a batch of work, so they remain in the role for another batch, and that continues almost indefinitely?

Turning `IsLightweightServer` on for these server worked like a charm and everything could be processed on time again. Then (with the help of this issue) we found that the sliding expiry no longer worked, causing long running jobs to be presumed failed and re-spawned. The result is multiple instances of the same job run concurrently, so we have had to turn it off again.

tl:dr: Split settings would give us the best of both worlds.

**Comment by davewebscale** [comment]
Thanks for looking into this @odinserj 
Linking the [fix](https://github.com/HangfireIO/Hangfire/commit/12c03316c8b8931943a85f1e1b8b4085aa0a7501) here
